app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -112,42 +193,127 @@ class PopupView : FrameLayout {
             return ViewHolder(binding)
         }
 
-        override fun onBindViewHolder(holder: ViewHolder, position: Int) =
-            holder.bind(getItem(position))
+        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+            val popupField: PopupField = getItem(position)
+            holder.setViewsBehavior(popupField)
+
+            holder.bind(popupField)
+        }
     }
 
     /**
      * Callback for calculating the diff between two non-null items in a list.
      */
-    private class DiffCallback : DiffUtil.ItemCallback<PopupAttribute>() {
+    private class DiffCallback : DiffUtil.ItemCallback<PopupField>() {
 
         override fun areItemsTheSame(
-            oldItem: PopupAttribute,
-            newItem: PopupAttribute
+            oldItem: PopupField,
+            newItem: PopupField
         ): Boolean {
             return oldItem == newItem
         }
 
         override fun areContentsTheSame(
-            oldItem: PopupAttribute,
-            newItem: PopupAttribute
+            oldItem: PopupField,
+            newItem: PopupField
         ): Boolean {
-            return oldItem.field == newItem.field && oldItem.value == newItem.value
+            return oldItem.fieldName == newItem.fieldName
         }
     }
 
     /**
      * The PopupAttributeListAdapter ViewHolder.
      */
-    private class ViewHolder(private val binding: ViewDataBinding) :
+    private inner class ViewHolder(private val binding: ViewDataBinding) :
         RecyclerView.ViewHolder(binding.root) {
 
+        val popupFieldValue: TextView by lazy {
+            binding.root.popupFieldValue
+        }
+
+        val popupFieldEditValue: EditText by lazy {
+            binding.root.popupFieldEditValue
+        }
+
         fun bind(
-            popupAttribute: PopupAttribute
+            popupField: PopupField
         ) {
-            binding.setVariable(BR.popupAttribute, popupAttribute)
+            binding.setVariable(BR.popupField, popupField)
+            binding.setVariable(BR.popupManager, popupManager)
             binding.executePendingBindings()
         }
+
+        /**
+         * Toggles the view for popup field value from edittext to textview and vice-versa, given the
+         * edit mode of the popupView.
+         */
+        fun setViewsBehavior(popupField: PopupField) {
```

Comment:
Would `updateViewMode` be more descriptive?

Reply:
updated.